File: Formula1/src/Formula1/JAVA/F1.java
package Formula1.JAVA;

public class F1 extends Monoplaza {
    private String nombrePatrocinador;

    
    ////////////////CONSTRUCTOR/////////////
    public F1(String nombrePiloto, String nacionalidad, int numeroPlaza, String escuderia, int cantidadPuntos, String pNombrePatrocinador) {
        super(nombrePiloto, nacionalidad, numeroPlaza, escuderia, cantidadPuntos);
        this.nombrePatrocinador = pNombrePatrocinador;
    }

    ///////////////////GETTERS Y SETTERS///////////////////
    public String getNombrePatrocinador() {
        return nombrePatrocinador;
    }
    public void setNombrePatrocinador(String nombrePatrocinador) {
        this.nombrePatrocinador = nombrePatrocinador;
    }
    ///////////METODOS////////////
    @Override
    public boolean posicionValida(int posicion) {
        if(posicion >=1 && posicion<=22){
        return true;
    }else{
        return false;
    } 
    }   


    public void otorgarPuntos(int posicion, boolean vueltaRapida) {
        if (!posicionValida(posicion)) return;

        switch (posicion) {
            case 1 -> cantidadPuntos += 25;
            case 2 -> cantidadPuntos += 18;
            case 3 -> cantidadPuntos += 15;
            case 4 -> cantidadPuntos += 12;
            case 5 -> cantidadPuntos += 10;
            case 6 -> cantidadPuntos += 8;
            case 7 -> cantidadPuntos += 6;
            case 8 -> cantidadPuntos += 4;
            case 9 -> cantidadPuntos += 2;
            case 10 -> cantidadPuntos += 1;
            default -> cantidadPuntos += 0;
        }
        if(vueltaRapida && posicion <=10){
            cantidadPuntos = cantidadPuntos+1;
        }
    }
    @Override
    public String toString() {
        return super.toString() + " | Patrocinador: " + nombrePatrocinador;
    }
}

File: Formula1/bin/Formula1/F1.php
<?php
namespace Formula1;

class F1 extends Monoplaza {
    private $nombrePatrocinador;

     ////////CONSTRUCTOR////////
    public function __construct(string $nombrePiloto, string $nacionalidad, int $numeroPlaza, string $escuderia, int $cantidadPuntos, string $nombrePatrocinador) {
        parent::__construct($nombrePiloto, $nacionalidad, $numeroPlaza, $escuderia, $cantidadPuntos);
        $this->nombrePatrocinador = $nombrePatrocinador;
    }

     // GETTERS Y SETTERS ////
    public function getNombrePatrocinador(){
    return $this->nombrePatrocinador;
    }
    public function setNombrePatrocinador(string $nombrePatrocinador){
    $this->nombrePatrocinador = $nombrePatrocinador;
    }

     // METODOS ///
    public function posicionValida($posicion){
        if ($posicion < 1) {
        return false;
        }
        if ($posicion > 22) {
        return false;
        }
        return true;
    }

    public function otorgarPuntos($posicion,$vueltaRapida){
        if ($posicion < 1 || $posicion > 22) {
        return false;
        }
        switch ($posicion) {
        case 1:
            $this->cantidadPuntos += 25;
            break;
        case 2:
            $this->cantidadPuntos += 18;
            break;
        case 3:
            $this->cantidadPuntos += 15;
            break;
        case 4:
            $this->cantidadPuntos += 12;
            break;
        case 5:
            $this->cantidadPuntos += 10;
            break;
        case 6:
            $this->cantidadPuntos += 8;
            break;
        case 7:
            $this->cantidadPuntos += 6;
            break;
        case 8:
            $this->cantidadPuntos += 4;
            break;
        case 9:
            $this->cantidadPuntos += 2;
            break;
        case 10:
            $this->cantidadPuntos += 1;
            break;
        default:
            break;
        }

        if ($vueltaRapida && $posicion <= 10) {
            $this->cantidadPuntos += 1;
        }
    }

    ///TOSTRING///
    public function __toString(){
    return parent::__toString() . "Patrocinador:" . $this->nombrePatrocinador;
    }
}